pkg/loader/translate_test.go

```diff
@@ -0,0 +1,134 @@
+
+	. "github.com/pingcap/check"
+)
+
+type slaveBinlogToTxnSuite struct{}
+
+var _ = Suite(&slaveBinlogToTxnSuite{})
+
+func (s *slaveBinlogToTxnSuite) TestTranslateDDL(c *C) {
+	db, table := "test", "hello"
+	sql := "CREATE TABLE hello (id INT AUTO_INCREMENT) PRIMARY KEY(id);"
+	binlog := pb.Binlog{
+		Type: pb.BinlogType_DDL,
+		DdlData: &pb.DDLData{
+			SchemaName: &db,
+			TableName:  &table,
+			DdlQuery:   []byte(sql),
+		},
+	}
+	txn := SlaveBinlogToTxn(&binlog)
+	c.Assert(txn.DDL.Database, Equals, db)
+	c.Assert(txn.DDL.Table, Equals, table)
+	c.Assert(txn.DDL.SQL, Equals, sql)
+}
+
+func (s *slaveBinlogToTxnSuite) TestTranslateDML(c *C) {
+	db, table := "test", "hello"
+	var oldVal, newVal int64 = 41, 42
+	dml := pb.DMLData{
+		Tables: []*pb.Table{
+			{
+				SchemaName: &db,
+				TableName:  &table,
+				ColumnInfo: []*pb.ColumnInfo{
+					{Name: "uid"},
+				},
+				Mutations: []*pb.TableMutation{
+					{
+						Type: pb.MutationType_Update.Enum(),
+						Row: &pb.Row{
+							Columns: []*pb.Column{
+								{Int64Value: &newVal},
+							},
+						},
+						ChangeRow: &pb.Row{
+							Columns: []*pb.Column{
+								{Int64Value: &oldVal},
+							},
+						},
+					},
+					{
+						Type: pb.MutationType_Insert.Enum(),
+						Row: &pb.Row{
+							Columns: []*pb.Column{
+								{Int64Value: &newVal},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	binlog := pb.Binlog{
+		DmlData: &dml,
+	}
+	txn := SlaveBinlogToTxn(&binlog)
+	c.Assert(txn.DMLs, HasLen, 2)
+	for _, dml := range txn.DMLs {
+		c.Assert(dml.Database, Equals, db)
+		c.Assert(dml.Table, Equals, table)
+	}
+	update := txn.DMLs[0]
+	c.Assert(update.Tp, Equals, UpdateDMLType)
+	c.Assert(update.Values, HasLen, 1)
+	c.Assert(update.Values["uid"], Equals, newVal)
+	c.Assert(update.OldValues, HasLen, 1)
+	c.Assert(update.OldValues["uid"], Equals, oldVal)
+	insert := txn.DMLs[1]
+	c.Assert(insert.Values, HasLen, 1)
+	c.Assert(insert.Values["uid"], Equals, newVal)
+}
+
+type columnToArgSuite struct{}
+
+var _ = Suite(&columnToArgSuite{})
+
+func (s *columnToArgSuite) TestHandleMySQLJSON(c *C) {
+	colVal := `{"key": "value"}`
+	arg := columnToArg("json", &pb.Column{BytesValue: []byte(colVal)})
+	c.Assert(arg, Equals, colVal)
+}
+
+func (s *columnToArgSuite) TestGetCorrectArgs(c *C) {
+	isNull := true
+	col := &pb.Column{IsNull: &isNull}
+	c.Assert(columnToArg("", col), IsNil)
+
+	var i64 int64 = 666
+	col = &pb.Column{Int64Value: &i64}
+	c.Assert(columnToArg("", col), Equals, i64)
+
+	var u64 uint64 = 777
+	col = &pb.Column{Uint64Value: &u64}
+	c.Assert(columnToArg("", col), Equals, u64)
+
+	var d float64 = 3.14
+	col = &pb.Column{DoubleValue: &d}
+	c.Assert(columnToArg("", col), Equals, d)
+
+	var b []byte = []byte{1, 2, 3}
+	col = &pb.Column{BytesValue: b}
+	c.Assert(columnToArg("", col), DeepEquals, b)
+
+	var ss string = "hello world"
+	col = &pb.Column{StringValue: &ss}
+	c.Assert(columnToArg("", col), DeepEquals, ss)
```

Comment:
For consistency with other literals 😄 

```suggestion
	c.Assert(columnToArg("", col), Equals, ss)
```

Maybe change Line 129 to use `BytesEquals` too.